### PR TITLE
CI-CD Deployment scripting for GitLab

### PR DESCRIPTION
**Pull request by xbrianh**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  DSS_TEST_ES_PATH: ./elasticsearch-${ES_VERSION}/bin/elasticsearch
+  DSS_ES_TIMEOUT: 30
+  DSS_UNITTEST_OPTS: "-v"
+  TERRAFORM_APPLY_ARGS: "-auto-approve"
+
+stages:
+  - deploy_dev
+
+before_script:
+  - apt-get update -q -y
```
Comment (hannes-ucsc) [inline .gitlab-ci.yml]: Suggested: I'd prefer the long-forms of these switches. For readability.

File: scripts/should_deploy.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    for link in linktext.split(","):
+        link_dest = link.split(";")[1]
+        if "next" in link_dest:
+            return link.split("<")[1].split(">")[0] 
+
+    return None
+
+
+def max_items(max_items=50):
+    def decorate(func):
```
Comment (hannes-ucsc) [inline scripts/should_deploy.py]: Suggested: I always use https://docs.python.org/3.6/library/functools.html#functools.wraps here.

File: scripts/should_deploy.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        for c in self.commits(branch):
+            return c['id']
+
+
+def should_deploy_dev():
+    owner = "HumanCellAtlas"
+    repo = "data-store"
+    branch = "master"
+
+    GH = GitHub(owner, repo)
```
Comment (hannes-ucsc) [inline scripts/should_deploy.py]: Suggest: Not a constant.

File: scripts/should_deploy.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+    integration_build_state = None
+    unit_tests_build_state = None
+    for b in Travis(owner, repo).builds_for_commit(branch, github_latest):
+        if b['commit']['message'].startswith("Integration"):
+            integration_build_state = b['state']
+        else:
+            unit_tests_build_state = b['state']
+
+        if integration_build_state is None and unit_tests_build_state is None:
```
Comment (hannes-ucsc) [inline scripts/should_deploy.py]: This looks odd. Don't you want to break out if both of them are *not* None?

File: .gitlab-ci.yml (hunk 1 above)
Comment (xbrianh) [reply .gitlab-ci.yml]: Done.

File: scripts/should_deploy.py (hunk 3 above)
Comment (xbrianh) [reply scripts/should_deploy.py]: But there aren't any constants in Python.. changed

File: scripts/should_deploy.py (hunk 4 above)
Comment (xbrianh) [reply scripts/should_deploy.py]: Insufficient coffee

**Pull request merged**